src/WebViewShared.tsx

```diff
@@ -47,22 +56,33 @@ const createOnShouldStartLoadWithRequest = (
     lockIdentifier: number,
   ) => void,
   originWhitelist: readonly string[],
+  deepLinkWhitelist: readonly string[],
   onShouldStartLoadWithRequest?: OnShouldStartLoadWithRequest,
 ) => {
   return ({ nativeEvent }: ShouldStartLoadRequestEvent) => {
     let shouldStart = true;
     const { url, lockIdentifier, isTopFrame } = nativeEvent;
 
     if (!_passesWhitelist(compileWhitelist(originWhitelist), url)) {
-      Linking.canOpenURL(url).then((supported) => {
-        if (supported && isTopFrame && /^https:\/\//.test(url)) {
-          return Linking.openURL(url);
-        }
-        console.warn(`Can't open url: ${url}`);
-        return undefined;
-      }).catch(e => {
-        console.warn('Error opening URL: ', e);
-      });
+      const foundMatchInBlocklist = matchWithRegexList(defaultDeeplinkBlocklist, url)
+
+      const _deeplinkWhitelist = (deepLinkWhitelist || []).map(stringWhitelistToRegex)
```

Comment:
I think this one is more fine since it would optimize the "sad" codepath that doesn't happen often, by slowing down the happy code path. It doesn't make sense to aalways precompute this whitelist when in reality we only need it 1% of the time.

Reply:
Makes sense, I agree.